### ISSUE TITLE
GDscript: wrong method argument type parsing in release export

### ISSUE THREAD

**Issue by vmikhav** [godotengine/godot#69532]
### Godot version

4.0 beta 7

### System information

Windows 10

### Issue description

I have a method which uses enum as an argument type:
```gdscript
func navigate(direction: TileSet.CellNeighbor):
```
and I call it in the next way:
```gdscript
level.navigate(TileSet.CELL_NEIGHBOR_LEFT_SIDE)
```

I works fine in editor or with debug export, but in release export I got next error:
```
SCRIPT ERROR: Parse Error: Invalid argument for "navigate()" function: argument 1 should be TileSet but is int.
```

### Steps to reproduce

Create method with argument with enum type (I used `TileSet.CellNeighbor`) and call it. Run project in release version.

### Minimal reproduction project

[GDscriptIssueTest.zip](https://github.com/godotengine/godot/files/10146800/GDscriptIssueTest.zip)

**Comment by vmikhav** [comment]
Updated

**Comment by anvilfolk** [comment]
So it appears that in release mode, Godot doesn't store `enum` names in classes. It only does this in debug mode. The fix is to allow it to do that for release mode, but that's a potentially big change which likely has some (minor?) performance downgrades.

That being said, I think if we want good typing & typechecking, then it makes sense to that.

**Comment by anvilfolk** [comment]
I don't have a release export available right now, but this ought to be fixed by #69532 if anybody can try!

**Comment by anvilfolk** [comment]
Getting this on `master`, so I'd closed this as fixed by #64253 :)

![image](https://user-images.githubusercontent.com/1133892/211415051-2acb09f1-bcf2-44b5-9eec-980424c2c537.png)

**Comment by adamscott** [comment]
CC @akien-mga for @anvilfolk's previous comment.
